_pages/typecheck.md

```diff
@@ -684,3 +684,358 @@ Cyclic module dependencies can cause problems for the type checker.  In order to
 ```lua
 local myModule = require(MyModule) :: any
 ```
+
+## Type functions
+
+Type functions are functions that operate on types instead of runtime values. They can use the [types](#types-library) library to transform existing types or create new ones.
```

Comment:
```suggestion
Type functions are functions that run during analysis time and operate on types, instead of runtime values. They can use the [types](#types-library) library to transform existing types or create new ones.
```